tekton-pipelines/base/tasks/slack-notification-task.yaml

```diff
@@ -54,10 +62,28 @@ spec:
         #!/usr/bin/env sh
         echo $(python get-thread | jq ".thread" -r) | tee $(results.thread.path)
 
+    - name: get-failed-info
+      image: bitnami/kubectl:latest
+      script: |
+        #!/bin/sh
+
+        pipelinerun_name=$(awk -F= '/tekton.dev\/pipelineRun/ {gsub(/"/, "", $2);print $2}' /etc/podinfo/labels)
+        kubectl -n ci get pipelinerun $pipelinerun_name -o json | jq '[.status.taskRuns | to_entries[] | select ( .value.status.conditions[0].reason == "Failed") | { task: .value.pipelineTaskName, pod: .value.status.podName, steps: .value.status.steps[] } | select (.steps.terminated.reason == "Error")][0] | { task: .task, pod: .pod, step: .steps.name }' > $(results.failed_info.path)
```

Comment:
pls break it into multiline, so it's easier to see it without horizontal scrolling

Reply:
what happens if permissions are not granted (or overwritten) and this step exits with non zero error code?